client/app/components/routes/operations/form/Operation.tsx

```diff
@@ -194,6 +202,54 @@ export default async function Operation({ numRow }: { numRow?: number }) {
     operation &&
     [Status.REJECTED, Status.APPROVED].includes(operation?.status as Status);
 
+  let userProfileFormData: UserProfileFormData | { error: string } =
+    await getUserFormData();
+
+  const formData = {
+    ...userProfileFormData,
+    ...operation,
+  };
+
+  const userEmail = (userProfileFormData as UserProfileFormData)?.email;
+  const applicationLeadEmail = formData?.application_lead?.email;
+  // If the current user is the application lead, we want to show the application lead fields
+  const isUserApplicationLead =
+    userEmail === applicationLeadEmail && applicationLeadEmail !== undefined;
+
```

Comment:
Wasn't entirely sure if this was enough for the check. I figured that if they didn't click the button but entered the same email there was just a misunderstanding so we would check this off. 

Also I tried to do this with a function similar to the `def user_is_senior_officer` in the `UserOperator` model and then a resolver in the schema but since we don't have access to `User` unlike in that table I decided to take care of this on the front end.

Reply:
Totally works for now, we probably want to review this implementation later to see if there is a best practice or standard that we can follow for future conditionals. I've added a tech debt card #602 to review this and start a conversation.